Playwright setup

lipmaneth opened the pull request.

ivelin commented on .github/workflows/playwright:
```diff
@@ -0,0 +1,15 @@
+  setup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies for all browsers
```
See the comment about redundant dependency installs on PR #14 .
Also the related comment about running actual tests as part of the CI script.

lipmaneth replied:
From my friend chatgpt:

_Based on the code provided, there does not seem to be any redundant dependency installs.

The first step installs the dependencies for all browsers using the command npx playwright install-deps.

The second step installs the dependencies for Chromium only using the command npx playwright install-deps chromium. This is necessary because not all browsers require the same dependencies, and installing dependencies for all browsers may result in unnecessary downloads and installation of packages.

Therefore, each step is installing the necessary dependencies for the intended browser and there does not seem to be any redundant installations_.

lipmaneth replied:
Can you please elaborate on this? "Also the related comment about running actual tests as part of the CI script."

I don't follow.

ivelin replied:
OK, looks like chatgpt is correct. No need to change anything:
https://playwright.dev/docs/cli#install-system-dependencies

ivelin commented:
#13  and #14 are duplicates. Keep one, close the other.

ivelin reviewed (changes requested):
Cool. See the other comment about redundant dependency installs.

ivelin reviewed (changes requested):
Just noticed the file name does not have the required .yaml extension to run.

ivelin reviewed (changes requested):
OK, looks like no duplicate install after all. 

Let's move on to adding the first GuardianUI e2e test to the CI run.

ivelin commented on .github/workflows/playwright.yaml:
```diff
@@ -0,0 +1,16 @@
+  setup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies for all browsers
```
@lipmaneth , very low priority, but there seems to be inconsistency in the playwright docs. It is not clear how the latest version behaves. When you add an actual test and get it to run, try to run the CI with and without this section in the action script. 

```
   run: npx playwright install-deps
```

I am not convinced that it's necessary based on the latest docs. These dependency installs can take a few minutes to run, which is why I like to keep them to a minimum.

lipmaneth replied:
Ok can take a look.

lipmaneth commented:
@ivelin can you take a look at this? I've set up the YAML file, test directory, and added your test from the app. All the code looks to be correct and configured correctly, but the checks are not successful.  Seemingly timing out for some reason?

Updated 1:09p CST: You can ignore the comment here. I'm still troubleshooting.

ivelin commented on tests/home-page-check:
```diff
@@ -0,0 +1,9 @@
+import { test } from "@framework/test";
```
@lipmaneth playwright test files should have a .js or .ts extension. They are either javascript or typescript code.

ivelin replied:
@lipmaneth see @0xLienid 's comment about using the base `@playwright/test` framework instead of the proprietary guardianui `@framework/test`.

ivelin commented on tests/home-page-check:
```diff
@@ -0,0 +1,9 @@
+import { test } from "@framework/test";
+
+test.describe("Verify GuardianUI.com home page", () => {
+	test("Should verify that the target page loads", async({ page, dappPage }) => {
```
@lipmaneth see comment about dropping `dappPage` in order to avoid pulling the proprietary test framework.

ivelin commented on tests/package.json:
```diff
@@ -0,0 +1,32 @@
+{
```
@lipmaneth `package.json` normally resides in the main project directory, not under `tests/`.

0xLienid replied:
Looks like this was just copied from the PlaywrightFramework repo. The better way is to use yarn or npm and initialize a new package.json in the main directory. Happy to jump on a call to walk you through that or just do it and push it up

ivelin commented on tests/package.json:
```diff
@@ -0,0 +1,32 @@
+    "axios": "^1.2.2",
+    "dotenv": "^16.0.3",
+    "dotenv-webpack": "^8.0.1",
+    "ethers": "^5.7.2",
```
@lipmaneth we will not need and of the `eth*` packages for this simple web2 site testing demo. You can remove them all from this file.

ivelin reviewed (changes requested):
See inline comments.

The playwritght yaml script fails because it is looking for `package.json` in the main (root) project directory, but you tucked it under `tests/`.

0xLienid commented on .github/workflows/playwright.yaml:
```diff
@@ -0,0 +1,18 @@
+      - name: Install dependencies for Chromium only
+        run: npx playwright install-deps chromium
+      - name: Run tests
+        run: npm run test -- --testPathPattern=tests
```
I tend to have better luck using yarn over npm. very low priority, but you could switch this to `yarn run test -- --testPathPattern=tests`

ivelin commented:
Closing as redundant to an already merged PR #15
